src/intervals/stream_buffer.jl

```diff
@@ -0,0 +1,94 @@
+
+
+@doc """
+A buffer in which pushing to the back, popping from the front (shift),
+and indexing is efficient.
+
+It's constructed as a circular buffer that resizes when it runs out of space.
+""" ->
+type StreamBuffer{T}
+    data::Vector{T}
+    first_idx::Int
+    last_idx::Int
+
+    function StreamBuffer()
+        return new(Array(T, 16), 1, 0)
+    end
+end
+
+
+function Base.length{T}(buf::StreamBuffer{T})
```

Comment:
This generic function is imported explicit on this module, therefore we don't need `Base.` here.